Source/Injector/Hooks/DarkSouls2/DS2_ReplaceServerAddressHook.cpp

```diff
@@ -99,10 +107,22 @@ bool DS2_ReplaceServerAddressHook::PatchKey(Injector& injector)
         {
             // If the memory is not writable yet, modify its protection (steam drm fucks with the protection during boot).
             MEMORY_BASIC_INFORMATION info;
-            if (VirtualQuery((void*)key, &info, sizeof(info)) == 0 ||
-                ((info.Protect & PAGE_READWRITE) == 0 && (info.Protect & PAGE_EXECUTE_READWRITE) == 0))
+            // If the programm is running on wine
+            if(winePrefix != nullptr)
             {
-                continue;
+                Log("you are using wine");
+                if (VirtualQuery((void*)key, &info, sizeof(info)) == 0) //wine doesn't emulate memory seafe features of windows {il TroncoNinja e' stato qui}
```

Comment:
I'm curious, what does wine actually return for the protection flags? I would expect it to return the same values as windows, as I can see a whole slew of compatibility issues if that wasn't the case.

Reply:
the Flags are all correct, as far as my testing goes, the problem is in the info struct, more specifically the changes made in the info.Protect (in windows info.Protect=64, in wine/proton info.Protect = 128). The memory seems to have the PAGE_EXECUTE_WRITECOPY flag instead of PAGE_EXECUTE_READWRITE